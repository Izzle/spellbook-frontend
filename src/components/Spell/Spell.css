.Spell__main {
    width: 300px;
    background-color: darkslategray;
    border: 1px solid black;
    margin-right: 10px;
}