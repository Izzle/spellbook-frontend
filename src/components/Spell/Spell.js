import React from 'react';

export default function Spell(props) {
    return (
      <div className='Spell__main'>
          <h4>{props.name}</h4>
          <p>{`Level: ${props.level} // School: Evocation // Range: 150 feet`}</p>
          <p>Classes: Cleric, Sorcerer, Wizard, Warlock</p>
          <p>Casting Time: 1 Action // Duration: Instantaneous // Components: V, S, M </p>
          <p>Description: boom fire</p>
          <h5>At higher levels</h5>
          <p>Description: Hits even harder</p>

          {/* key={spell.id}
                name={spell.spell_name}
                classes={spell.classes}
                level={spell.level}
                school={spell.school}
                range={spell.range}
                castingTime={spell.castingTime}
                spellComponents={spell.components}
                duration={spell.duration}
                description={spell.description}
                higherLevels={spell.higherLevels} */}
      </div>
    )
}