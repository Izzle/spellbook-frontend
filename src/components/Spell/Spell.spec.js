import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Spell from './Spell';
import { BrowserRouter } from 'react-router-dom';


describe('<Spell />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Spell higherLevels={'Plus 1 damage'} /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><Spell higherLevels={'Plus 1 damage'}/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});