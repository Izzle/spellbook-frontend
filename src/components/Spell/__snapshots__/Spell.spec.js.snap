// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spell /> renders this UI as expected 1`] = `
<div
  className="Spell__main"
  onClick={[Function]}
>
  <h4 />
  <p>
    Level: undefined // School: undefined // Range: undefined
  </p>
  <p>
    Classes: undefined (FIX THIS: Currently using the class ID)
  </p>
  <p>
    Casting Time: undefined // Duration: undefined // Components: undefined
  </p>
  <p>
    Description: undefined
  </p>
  <h5>
    At higher levels
  </h5>
  <p>
    Plus 1 damage
  </p>
</div>
`;
